library/common/glfw/embedder.cc

```diff
@@ -80,6 +80,12 @@ struct FlutterEmbedderState {
   double window_pixels_per_screen_coordinate = 1.0;
 };
 
+// Struct for storing state of a Flutter engine instance.
+struct FlutterEngineState {
+  // The handle to the Flutter engine instance.
+  FlutterEngine engine;
+};
```

Comment:
Longer term we'd probably want to have the plugin handlers and the message dispatcher in this struct (to allow for plugins in a headless process), and then have it as a member of the other struct. We can wait and see how much of this code survives a switch to a full Linux and Windows implementation of the shell though, since it may never matter; just mentioning for history in case we're trying to remember later why it's separate :)

Reply:
I actually started that way, but I wondered if it would make the code a bit more complex. I'll just add it now so we don't have to do it later.

Reply:
I tried this way but I'm bumping into a few issues. I'll make a new branch and submit it later. Will keep this way for now.